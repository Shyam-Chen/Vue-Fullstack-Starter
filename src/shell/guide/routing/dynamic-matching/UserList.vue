<template>
  <v-layout row>
    <div>
      <v-list dense class="elevation-1">
        <template v-for="item in items">
          <v-subheader v-if="item.header" :key="item.header">{{ item.header }}</v-subheader>

          <v-list-tile v-else :key="item.title" avatar @click="$router.push(`/guide/routing/dynamic-matching/${item.title}`)">
            <v-list-tile-avatar>
              <img :src="item.avatar">
            </v-list-tile-avatar>
            <v-list-tile-content>
              <v-list-tile-title v-html="item.title"></v-list-tile-title>
              <v-list-tile-sub-title v-html="item.subtitle"></v-list-tile-sub-title>
            </v-list-tile-content>
          </v-list-tile>
        </template>
      </v-list>
    </div>
  </v-layout>
</template>

<script>
export default {
  data() {
    return {
      items: [
        { header: 'User List' },
        { avatar: 'https://vuetifyjs.com/static/doc-images/lists/1.jpg', title: 'Jason Oner' },
        { avatar: 'https://vuetifyjs.com/static/doc-images/lists/2.jpg', title: 'Travis Howard' },
        { avatar: 'https://vuetifyjs.com/static/doc-images/lists/3.jpg', title: 'Ali Connors' },
        { avatar: 'https://vuetifyjs.com/static/doc-images/lists/4.jpg', title: 'Cindy Baker' },
        { avatar: 'https://vuetifyjs.com/static/doc-images/lists/5.jpg', title: 'Britta Holt' },
      ],
    };
  },
};
</script>

<style scoped>
</style>
