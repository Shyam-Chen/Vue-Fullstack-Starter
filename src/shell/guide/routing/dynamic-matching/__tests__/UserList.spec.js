import { shallowMount } from '@vue/test-utils';

import UserList from '../UserList';

describe('UserList', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(UserList);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
