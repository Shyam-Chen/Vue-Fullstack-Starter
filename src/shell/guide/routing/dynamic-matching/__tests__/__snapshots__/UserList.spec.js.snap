// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList should render an initial component 1`] = `
<v-layout-stub tag="div" row="">
  <div>
    <v-list-stub tag="div" tile="true" dense="true" class="elevation-1">
      <v-subheader-stub>User List</v-subheader-stub>
      <v-list-item-stub activeclass="" tag="div" avatar="">
        <v-list-item-avatar-stub size="40"><img src="https://vuetifyjs.com/static/doc-images/lists/1.jpg"></v-list-item-avatar-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub></v-list-item-title-stub>
          <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div" avatar="">
        <v-list-item-avatar-stub size="40"><img src="https://vuetifyjs.com/static/doc-images/lists/2.jpg"></v-list-item-avatar-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub></v-list-item-title-stub>
          <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div" avatar="">
        <v-list-item-avatar-stub size="40"><img src="https://vuetifyjs.com/static/doc-images/lists/3.jpg"></v-list-item-avatar-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub></v-list-item-title-stub>
          <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div" avatar="">
        <v-list-item-avatar-stub size="40"><img src="https://vuetifyjs.com/static/doc-images/lists/4.jpg"></v-list-item-avatar-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub></v-list-item-title-stub>
          <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div" avatar="">
        <v-list-item-avatar-stub size="40"><img src="https://vuetifyjs.com/static/doc-images/lists/5.jpg"></v-list-item-avatar-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub></v-list-item-title-stub>
          <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-list-stub>
  </div>
</v-layout-stub>
`;
